inference-engine/src/gna_plugin/gna_executable_network.hpp

```diff
@@ -67,6 +68,28 @@ class GNAExecutableNetwork : public InferenceEngine::ExecutableNetworkThreadSafe
         THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str;
     }
 
+    void SetConfig(const std::map<std::string, InferenceEngine::Parameter>& config,
+                   InferenceEngine::ResponseDesc* /* resp */) override {
+        using namespace InferenceEngine::GNAConfigParams;
+        if (config.empty()) {
+            THROW_IE_EXCEPTION << "The list of configuration values is empty";
+        }
+        for (auto&& item : config) {
+            if (item.first != KEY_GNA_DEVICE_MODE) {
+                THROW_IE_EXCEPTION << "The following config value cannot be changed dynamically for ExecutableNetwork in the GNA plugin: "
+                                   << item.first << ". Only " << KEY_GNA_DEVICE_MODE << " is supported.";
+            }
+        }
+        IE_ASSERT(config.begin()->first == KEY_GNA_DEVICE_MODE);
+        auto mode = config.begin()->second.as<std::string>();
+        if (mode == InferenceEngine::GNAConfigParams::GNA_SW_FP32) {
```

Comment:
should we also check that switching from gna_spw_fp32 mode to any other shouldn't work?

Reply:
Right! I wanted to do it but forgot, thanks. 👍

Reply:
Fixed, test added